cdap-ui/app/cdap/components/FastAction/SetPreferenceAction/SetPreferenceModal.js

```diff
@@ -0,0 +1,430 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, {Component, PropTypes} from 'react';
+import isObject from 'lodash/isObject';
+import upperFirst from 'lodash/upperFirst';
+import orderBy from 'lodash/orderBy';
+import T from 'i18n-react';
+import shortid from 'shortid';
+import { Modal, ModalHeader, ModalBody } from 'reactstrap';
+import myPreferenceApi from 'api/preference';
+import {convertProgramToApi} from 'services/program-api-converter';
+import KeyValuePairs from 'components/KeyValuePairs';
+import NamespaceStore from 'services/NamespaceStore';
+require('./SetPreferenceAction.scss');
+
+export default class SetPreferenceAction extends Component {
```

Comment:
This should be `SetPreferenceActionModalContent` right? I'm not sure how react is able to differentiate between the two. The seem to be having the same name.